torch/csrc/jit/passes/quantization.cpp

```diff
@@ -818,13 +871,142 @@ void InsertObserversHelper::insertObservers(
       }
     }
   }
+}
+
+c10::optional<script::Module>
+InsertObserversHelper::getObserverFor(Value* v) {
+  if (observer_for_value_.count(v)) {
+    auto observer = observer_for_value_.at(v);
+    return observer;
+  }
+  c10::optional<script::Module> result;
+  if (boundary_value_map_.count(v)) {
+    for (Value* next : boundary_value_map_.at(v)) {
+      auto observer_opt = getObserverFor(next);
+      if (observer_opt) {
+        // Need to make sure all values are
+        // configured with same observer
+        if (result) {
+          TORCH_CHECK(
```

Comment:
Should this be an assertion? Isn't it possible to hit this case with a legal model with a lecal qconfig (i.e. if we have conv in two submodules but requested to quantize it in only one of them)?

Reply:
this is checking qconfig for values, the case would be

```
class M(torch.nn.Module):
  def __init__(self):
       self.conv = torch.nn.Conv2d(...)
       self.fc = torch.nn.Linear(...)
   def forward(self, x):
       x1 = self.conv(x)
       x2 = self.fc(x1)
       return x2
```

And we configure self.conv with QConfig1 and self.fc with QConfig2, so `x1` will have two different qconfigs in this case.